pkg/ipcache/metadata.go

```diff
@@ -361,10 +361,26 @@ func (ipc *IPCache) doInjectLabels(ctx context.Context, modifiedPrefixes []netip
 				break
 			}
 
+			// If the old entry is tainted, and we are attempting to overwrite it,
+			// then we want to keep its old source around in case we eventually
+			// stop managing it and need to restore the old source
+			var newOverwrittenLegacySource source.Source
+			switch {
+			case oldID.exclusivelyOwnedByLegacyAPI():
```

Comment:
oldid may be the zero value here, so probably better to put this behind a `if entryExists {` conditional?

Reply:
Yes. The old code wasn't doing that either, but this makes it very clear that something was off before. Still need to fix.